src/components/Chart/Bars.js

```diff
@@ -329,12 +336,28 @@ const BarChart = props => {
             : isLastSegment
             ? 'left'
             : 'right')
+
+        d.inlineLabel = [
+          inlineValue && xAxis.format(d.value),
+          inlineValueUnit && inlineValueUnit,
+          inlineLabel && d.datum[inlineLabel]
+        ].join(' ')
+        d.inlineLabelTextWidth = labelGauger(d.inlineLabelText)
+
+        const shiftInlineText = isLast && d.width <= d.inlineLabelTextWidth
```

Comment:
```suggestion
        d.shiftInlineText = bar.segments.length === 1 && d.width <= d.inlineLabelTextWidth
```
safe on segment for later use in rendering (ensuring the different color rule is only applied when shifted)

And restrict to single segment. Multi segment has bad edge cases even when last:
<img width="1224" alt="Screenshot 2021-04-09 at 00 16 47" src="https://user-images.githubusercontent.com/410211/114103940-4a29c800-98ca-11eb-9410-73c18ae44aa4.png">

Reply:
This would negate the fix in some cases see:
![Screenshot 2021-04-09 at 08 50 59](https://user-images.githubusercontent.com/20746301/114140728-f5f60680-9910-11eb-8f1e-483f95b4563d.jpg)
I think in this case it's actually a configuration issue. The bars can be configured to look right with (isLast)
![Screenshot 2021-04-09 at 08 59 23](https://user-images.githubusercontent.com/20746301/114141509-f6db6800-9911-11eb-94ba-ae15a52233d3.jpg)

Reply:
@ovbm found a solution which automatically does it:
https://github.com/orbiting/styleguide/pull/376/commits/ae32178ff3afb84753b876ee2457082be17c565e

Hopefully it does not have any regression.